### PR TITLE
bpo-37022: Fix bug where `pdb`'s `do_p`/`do_pp` commands swallow exceptions from `repr`

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://bugs.python.org/issue37022.

https://bugs.python.org/issue37022

### CODE REVIEW

File: Lib/pdb.py [hunk 1]
```diff
@@ -1176,23 +1173,31 @@ def _getval_except(self, arg, frame=None):
...
 
+    def _error_exc(self):
+        exc_info = sys.exc_info()[:2]
+        self.error(traceback.format_exception_only(*exc_info)[-1].strip())
+
+    def _msg_val_func(self, arg, func):
+        try:
+            val = self._getval(arg)
+        except:
+            return
```
Comment (iritkatriel) [inline Lib/pdb.py]: Might be worth a comment that getval already emitted an error

**Review by iritkatriel — APPROVED** [review]

**Comment by gvanrossum** [comment]
This looks to me like a bugfix and seems safe to backport to 3.9 and 3.8. Clearly the intention was to just catch the exception from `_getval()` and that it caught something else would be an unfortunate side effect, not something someone would rely on.

File: Lib/pdb.py (hunk 1 above)
Comment (iritkatriel) [inline Lib/pdb.py]:
```suggestion
            return  # _getval() has displayed the error
```

**Review by iritkatriel — APPROVED** [review]

**Comment by gvanrossum** [comment]
Irit, you can manage this by yourself, right?

**Comment by iritkatriel** [comment]
> Irit, you can manage this by yourself, right?

Yes. Just waiting for the bot to finish.

**Pull request merged**